pkg/labelsfilter/filter.go

```diff
@@ -206,6 +206,8 @@ func defaultLabelPrefixCfg() *labelPrefixCfg {
 		"!controller-revision-hash",     // ignore controller-revision-hash
 		"!annotation.*",                 // ignore all annotation labels
 		"!etcd_node",                    // ignore etcd_node label
+
+		"!io.cilium.k8s.namespace.labels.fluxcd.io/sync-gc-mark", // ignore hash from flux
```

Comment:
I don't think I follow this change. All the other filtered labels are either `special` labels in the context of k8s or cilium agent. Why are we adding custom labels here? How do we keep these labels in sync with the user environment? We could potentially expose an API/config option for users to add these exceptions, or introduce a reserved `io.cilium.ignorelabel.` string that users can prepend to the relevant labels.

Reply:
> All the other filtered labels are either special labels in the context of k8s control plane or cilium agent. Why are we adding custom labels here?

Not sure what you mean by special and custom here.

> How do we keep these labels in sync with the user environment?

I don't understand the question. This is the default list, there's no sync. needed AFAICS.

> We could potentially expose an API/config option for users to add these exceptions, or introduce a reserved io.cilium.ignorelabel. string that users can prepend to the relevant labels.

That's already the case; we have the `labels` cilium-agent option. This is changing the default value to that option.

>  I wonder if it's enough to filter/warn about all the labels that are unused in any of the applied CNPs?

I think that's what André is proposing. I'm closing this PR has André said we can't change the default list at all without risking user downtime. (Or maybe André is trying to scare us because he really wants to work on his new solution :smile: )